Megatron-LM-v1.1.5-ZeRO3/examples/ds_pretrain_gpt2-zero2_8.3B_params_16GPUs.sh

```diff
@@ -0,0 +1,166 @@
+MP_SIZE=8
+
+DEBUG=1
+if [[ ${DEBUG} == 1 ]];  then
+       MP_SIZE=8
+       NUM_WORKERS=1
+       NUM_GPUS_PER_WORKER=16
+       HIDDEN_SIZE=3072
+       NUM_ATTN_HEADS=24
+       NUM_LAYERS=72
+       BATCHSIZE=16
+else
+       NUM_WORKERS=${DLTS_NUM_WORKER}
+       NUM_GPUS_PER_WORKER=${DLTS_NUM_GPU_PER_WORKER}
+       HIDDEN_SIZE=8192
+       NUM_ATTN_HEADS=32
+       NUM_LAYERS=50
+       BATCHSIZE=4
+
+       #HIDDEN_SIZE=4096
+       #NUM_LAYERS=24 # 50
+       #BATCHSIZE=16
+fi
+
+
+#BASE_DATA_PATH=/data/Megatron-LM/data
+#DATA_PATH=${BASE_DATA_PATH}/indexed_datasets/megatron
+#VOCAB_PATH=${BASE_DATA_PATH}/gpt2-vocab.json
+#MERGE_PATH=${BASE_DATA_PATH}/gpt2-merges.txt
+DATA_PATH=/data/DeepSpeed_data/Megatron_wikipedia/my-gpt2_text_document
+VOCAB_PATH=/data/DeepSpeed_data/Megatron_wikipedia/gpt2-vocab.json
+MERGE_PATH=/data/DeepSpeed_data/Megatron_wikipedia/gpt2-merges.txt
+CHECKPOINT_PATH=checkpoints/gpt2_345m_ds
+
+script_path=$(realpath $0)
+script_dir=$(dirname $script_path)
+if [[ -z $1 ]]; then
+       config_json="$script_dir/ds_zero_stage_2_config_8.3B_params_16GPUs.json"
+
+       # offloads to NVMe
+       #config_json="$script_dir/ds_zero_stage_infinity_config.json"
+else
+       config_json=$script_dir/`basename $1`
+fi
+
+#ZeRO Configs
+stage=2
+reduce_scatter=true
+contigious_gradients=true
+rbs=50000000
+agbs=5000000000
+
+#Activation Checkpointing and Contigious Memory
+chkp_layers=1
+PA=true
+PA_CPU=true
+CC=true
+SYNCHRONIZE=true
+PROFILE=false
+
+# TiledLinear splits, 0 is disable
+TILED_LINEAR="false"
+TILE_DIM=1
+
+
+# Megatron Model Parallelism
+LOGDIR="tboard-zero2/stage${stage}-lazyscatter-${NUM_LAYERS}l_${HIDDEN_SIZE}h_${NUM_WORKERS}n_${NUM_GPUS_PER_WORKER}g_${MP_SIZE}mp_${BATCHSIZE}b"
+
+#--load $CHECKPOINT_PATH \
+gpt_options=" \
+        --model-parallel-size ${MP_SIZE} \
+        --num-layers $NUM_LAYERS \
+        --hidden-size $HIDDEN_SIZE \
+        --num-attention-heads ${NUM_ATTN_HEADS} \
+        --seq-length 1024 \
+        --max-position-embeddings 1024 \
+        --batch-size $BATCHSIZE \
+        --train-iters 320000 \
+        --lr-decay-iters 320000 \
+        --save $CHECKPOINT_PATH \
+        --data-path $DATA_PATH \
+        --vocab-file $VOCAB_PATH \
+        --merge-file $MERGE_PATH \
+        --data-impl mmap \
+        --split 949,50,1 \
+        --distributed-backend nccl \
+        --lr 1.5e-4 \
+        --lr-decay-style cosine \
+        --min-lr 1.0e-5 \
+        --weight-decay 1e-2 \
+        --clip-grad 1.0 \
+        --warmup 0.01 \
+        --checkpoint-activations \
+        --log-interval 1 \
+        --save-interval 10000 \
+        --eval-interval 2000 \
+        --eval-iters 10 \
+        --fp16 \
+        --scattered-embeddings \
+        --split-transformers \
+"
+        #--tensorboard-dir ${LOGDIR}
+
+ deepspeed_options=" \
+                --deepspeed \
+                --deepspeed_config ${config_json} \
+                --zero-stage ${stage} \
```

Comment:
Does this have the exact same effect as https://github.com/ROCmSoftwarePlatform/DeepSpeedExamples/pull/20/files#diff-10efc3ef681fa5df4e04cf177d380ce2cf883af28b326d949613e291fafd3e75R6?

Reply:
(https://github.com/ROCmSoftwarePlatform/DeepSpeedExamples/pull/20/files#diff-10efc3ef681fa5df4e04cf177d380ce2cf883af28b326d949613e291fafd3e75R6) overwrites the zero-stage given in https://github.com/ROCmSoftwarePlatform/DeepSpeedExamples/pull/20/files#diff-488666ef8a078822d4f9da40457dcd17c656f16ead22c5b4b0c41363d3b0d0d3R50